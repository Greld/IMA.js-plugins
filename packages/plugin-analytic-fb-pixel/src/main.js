import { ScriptLoaderPlugin } from 'ima-plugin-script-loader';
import FacebookPixelAnalytic from './FacebookPixelAnalytic';

let defaultDependencies = [
  ScriptLoaderPlugin,
  '$Window',
  '$Dispatcher',
  '$Settings.plugin.analytic.fbPixel'
];

let $registerImaPlugin = () => { };

let initBind = (ns, oc) => {
  oc.inject(FacebookPixelAnalytic, defaultDependencies);
};

let initSettings = () => {
  return {
    prod: {
      plugin: {
        analytic: {
          fbPixel: {
            id: null
          }
        }
      }
    },

    test: {},

    dev: {}
  };
};

export {
  FacebookPixelAnalytic,
  defaultDependencies,
  $registerImaPlugin,
  initBind,
  initSettings
};
