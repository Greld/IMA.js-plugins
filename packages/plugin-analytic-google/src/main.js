import { ScriptLoaderPlugin } from 'ima-plugin-script-loader';
import GoogleAnalytic from './GoogleAnalytic.js';

let defaultDependencies = [
  ScriptLoaderPlugin,
  '$Window',
  '$Dispatcher',
  '$Settings.plugin.analytic.google'
];

let $registerImaPlugin = () => {};

let initBind = (ns, oc) => {
  oc.inject(GoogleAnalytic, defaultDependencies);
};

let initSettings = () => {
  return {
    prod: {
      plugin: {
        analytic: {
          google: {
            service: 'UA-XXXXXXX-X',
            settings: {}
          }
        }
      }
    },

    test: {},

    dev: {}
  };
};

export {
  GoogleAnalytic,
  defaultDependencies,
  $registerImaPlugin,
  initBind,
  initSettings
};
