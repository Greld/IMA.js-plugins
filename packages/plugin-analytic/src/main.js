import AbstractAnalytic from './AbstractAnalytic';
import Events from './Events';

let $registerImaPlugin = () => {};

export { AbstractAnalytic, Events, $registerImaPlugin };
