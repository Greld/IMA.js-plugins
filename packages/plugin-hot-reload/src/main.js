import HotReloadService from './HotReloadService';

const $registerImaPlugin = () => {};

const initServices = (ns, oc) => {
  oc.get(HotReloadService).init();
};

export { initServices, $registerImaPlugin, HotReloadService };
