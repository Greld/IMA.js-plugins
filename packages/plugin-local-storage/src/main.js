import LocalStorage from './LocalStorage';

let $registerImaPlugin = () => {};

export { LocalStorage, $registerImaPlugin };
