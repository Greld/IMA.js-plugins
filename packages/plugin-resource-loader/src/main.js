import ResourceLoader from './ResourceLoader';

let $registerImaPlugin = () => {};

export default ResourceLoader;

export { ResourceLoader, $registerImaPlugin };
