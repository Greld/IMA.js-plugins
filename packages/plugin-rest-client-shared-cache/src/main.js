import SharedCache from './SharedCache';
import SharedCachePostProcessor from './SharedCachePostProcessor';
import SharedCachePreProcessor from './SharedCachePreProcessor';

export { SharedCache, SharedCachePostProcessor, SharedCachePreProcessor };
