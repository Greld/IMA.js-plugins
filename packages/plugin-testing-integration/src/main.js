import { setConfig } from './configuration';
import { initImaApp, clearImaApp } from './app';

export { setConfig, initImaApp, clearImaApp };
