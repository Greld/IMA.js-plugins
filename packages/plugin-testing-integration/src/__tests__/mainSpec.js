import * as main from '../main';

describe('Main', () => {
  it('can export all necessary properties', () => {
    expect(main).toMatchSnapshot();
  });
});
