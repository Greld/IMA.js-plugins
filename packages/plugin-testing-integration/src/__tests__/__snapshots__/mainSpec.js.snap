// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main can export all necessary properties 1`] = `
Object {
  "clearImaApp": [Function],
  "initImaApp": [Function],
  "setConfig": [Function],
}
`;
